in_toto/util.py

```diff
@@ -136,3 +136,49 @@ def prompt_generate_and_write_rsa_keypair(filepath, bits):
   generate_and_write_rsa_keypair"""
   password = prompt_password()
   generate_and_write_rsa_keypair(filepath, bits, password)
+
+def compare_dictionaries(dictionary_one, dictionary_two):
+  """
+  <Purpose>
+    Compares corresponding keys and corresponding values in order to determine changes
+    across two separate dictionaries
+
+  <Arguments>
+    dictionary_one
+      The first dictionary structure argument
+
+    dictionary_two
+      The second dictionary structure argument
+
+  <Returns>
+    A tuple value which contains dict information: (created, deleted, modified, unchanged)
+  """
+  
+  # Find newly created keys, comparatively
+  created = set(dictionary_one.keys())- set(dictionary_two.keys())
+  
+  # Find keys that were deleted, comparatively
+  deleted = set(dictionary_two.keys()) - set(dictionary_one.keys())
+
+  # Find intersection between dictionaries, comparatively
+  intersections = set(dictionary_one.keys()) & set(dictionary_two.keys())
+  modified = []
+  unchanged = []
+  for element in intersections:
+    if dictionary_one[element] == dictionary_two[element]:
+      # Add matched value to be  "unchanged" list
+      unchanged.append(element)
+    else:
+      # Add mismatched value to "modified" list
+      modified.append(element)
+
+  return created, deleted, modified, unchanged
+
+def check_string(string_arg):
+  """ Checks strings that are empty and defaults them to 'None' string"""
+  if not isinstance(string_arg, str):
+    return string_arg
+  else:
+    if not string_arg:
+      string_arg = "None"
+  return string_arg
```

Comment:
I'm not sure if a `"None"` string is really more meaningful than just displaying an empty string. So I guess we don't really need the `check_string` function.

Reply:
All mentions of this function (`check_string`) have been removed in the latest commit!